### PR TITLE
Enhance documentation

### PR DESCRIPTION
**Pull request by kouloumos**
Refined project documentation and added comments during an in-depth review to map out the project's architecture and component interactions. Open to feedback and further contributions.

### CODE REVIEW

**Review by Extheoisah — APPROVED** [review]

File: src/context/SearchQueryContext.tsx [hunk 1]
```diff
@@ -25,23 +25,49 @@ export type SearchQueryContextType = {
...
+ * 2. Utilizes useMemo to memoize calculations for search parameters and URLSearchParams for efficient query manipulation.
+ * 3. The `useSearch` hook fetches search results based on the current search parameters.
+ * 4. Provides two main functions, `makeQuery` and `handlePageChange`, to update the URL's query parameters
+ *    and thereby trigger new searches or page changes. These updates are performed through the NextJS router,
+ *    allowing for client-side navigation without full page reloads.
+ * 5. Any change to the URL's query parameters triggers a re-fetch of search results via `useSearch`,
+ *    updating the `queryResult` state with the new results.
+ * 6. The updated search state (current search query, search results, pagination information) is made available
+ *    to child components through the `SearchQueryContext`, enabling a reactive search UI that updates in response
+ *    to user actions and URL changes.
```
Comment (BalogunofAfrica) [inline src/context/SearchQueryContext.tsx:47]: These points are helpful, but I think they would be better placed in a general explanations file (markdown?) than a docstring which should give the reader and idea of the API/Interface of the component not some implementation details.

**Review by BalogunofAfrica — COMMENTED** [review]
Nice job on adding some documentation. But I find that asides the docstring (`/** */`) which can be used for quick inspection, the regular comments (`//`) are not really needed.

A good addition to the docstring would also be to include an example tag (`@example`) to illustrate the general API/Interaction with what component is been documented.

File: src/context/SearchQueryContext.tsx (hunk 1 above)
Comment (kouloumos) [reply src/context/SearchQueryContext.tsx:47]: In my opinion, context about the application must be in places that readers/developer will find it while trying to understand the implementation details of the codebase.
So I don't think that creating a markdown file somewhere else in order to explain what happens in the `SearchQueryProvider` is a good strategy.

**Comment by kouloumos** [comment]
> A good addition to the docstring would also be to include an example tag (@example) to illustrate the general API/Interaction with what component is been documented.

Could you clarify what you mean by an example tag? If you're suggesting an example of how to use this component, I'm not convinced of its utility. Typically, when I'm navigating new code, my challenge isn't in figuring out how to use a component but in understanding its role within the broader application. In my view, time spent on crafting examples could be better spent on writing detailed comments that aid developers in comprehending the complexities of the code.

**Pull request merged**